code/src/utils/filter-products-by-category/categoryFilter.ts

```diff
@@ -0,0 +1,15 @@
+import { Product } from "@/types/product.types";
+
+const categoryFilter = (
+  category: string | null,
+  productsList: Product[] | undefined
```

Comment:
I suggest adding comment for line `Product[] | undefined` that explains why we use `productsList: Product[] | undefined` instead of `productsList?: Product[]`

Reply:
I changed to syntax to what you suggested. I agree. that it looks much more readable.